src/cis/fam_cis_direct.cpp

```diff
@@ -1299,12 +1320,55 @@ Fam_Backup_Info Fam_CIS_Direct::get_backup_info(std::string BackupName,
     ostringstream message;
 
     CIS_DIRECT_PROFILE_START_OPS()
-    Fam_Memory_Service *memoryService = get_memory_service(memoryServerId);
+    auto obj = memoryServers->find(memoryServerId);
+    Fam_Memory_Service *memoryService = get_memory_service((uint64_t)obj->first);
     Fam_Backup_Info info = memoryService->get_backup_info(BackupName);
     CIS_DIRECT_PROFILE_END_OPS(cis_get_backup_info);
     return info;
 }
 
+std::string Fam_CIS_Direct::list_backup(std::string BackupName, uint64_t memoryServerId) {
+    ostringstream message;
```

Comment:
We will need permission checks for list_backup too. It shouldn't work if there is no permission.